localstack/services/sns/sns_listener.py

```diff
@@ -937,3 +975,33 @@ def check_filter_policy(filter_policy, message_attributes):
 
 def is_raw_message_delivery(susbcriber):
     return susbcriber.get("RawMessageDelivery") in ("true", True, "True")
+
+
+def store_delivery_log(
+    subscriber: dict, success: bool, message: str, message_id: str, delivery: dict = {}
```

Comment:
nit: let's better try to use immutable default argument values (`None`), and then assign a default value (`{}`) within the function body itself: (We can use the `not_none_or(..)` util from utils/common.py) 
```
def store_delivery_log(
    subscriber: dict, success: bool, message: str, message_id: str, delivery: dict = None
):
    delivery = not_none_or(delivery, {})
    ...
```